fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -1046,89 +1019,7 @@ private FHIRPersistenceNotSupportedException buildNotSupportedException(String m
-            if (existingResourceDTO.isDeleted()) {
-
-                addWarning(IssueType.DELETED, "Resource of type'" + resourceType.getSimpleName() +
-                        "' with id '" + logicalId + "' is already deleted.");
-
-                SingleResourceResult<T> result = new SingleResourceResult.Builder<T>()
-                        .success(true)
-                        .interactionStatus(InteractionStatus.READ)
-                        .resource(existingResource)
-                        .build();
-
-                return result;
-            }
-
-            int newVersionNumber = existingResourceDTO.getVersionId() + 1;
-            Instant lastUpdated = Instant.now(ZoneOffset.UTC);
-
-            // Update the soft-delete resource to reflect the new version and lastUpdated values.
-            T updatedResource = copyAndSetResourceMetaFields(existingResource, existingResource.getId(), newVersionNumber, lastUpdated);
-
-            // Create a new Resource DTO instance to represent the deleted version.
-            com.ibm.fhir.persistence.jdbc.dto.Resource resourceDTO =
-                    createResourceDTO(logicalId, newVersionNumber, lastUpdated, updatedResource,
-                        getResourcePayloadKeyFromContext(context));
-            resourceDTO.setDeleted(true);
-
-            // Persist the logically deleted Resource DTO.
-            resourceDao.setPersistenceContext(context);
-            resourceDao.insert(resourceDTO, null, null, null, IF_NONE_MATCH_NULL);
-
-            if (log.isLoggable(Level.FINE)) {
-                log.fine("Persisted FHIR Resource '" + resourceDTO.getResourceType() + "/" + resourceDTO.getLogicalId() + "' id=" + resourceDTO.getId()
-                            + ", version=" + resourceDTO.getVersionId());
-            }
-
-            SingleResourceResult<T> result = new SingleResourceResult.Builder<T>()
-                    .success(true)
-                    .interactionStatus(InteractionStatus.MODIFIED)
-                    .resource(updatedResource)
-                    .build();
-
-            return result;
-        }
-        catch(FHIRPersistenceFKVException e) {
-            throw e;
-        }
-        catch(FHIRPersistenceException e) {
-            throw e;
-        }
-        catch(Throwable e) {
-            FHIRPersistenceException fx = new FHIRPersistenceException("Unexpected error while performing a delete operation.");
-            log.log(Level.SEVERE, fx.getMessage(), e);
-            throw fx;
-        }
-        finally {
-            log.exiting(CLASSNAME, METHODNAME);
-        }
-    }
-
-    @Override
-    public <T extends Resource> void deleteWithMeta(FHIRPersistenceContext context, T resource) throws FHIRPersistenceException {
+    public <T extends Resource> void delete(FHIRPersistenceContext context, T resource) throws FHIRPersistenceException {
         final String METHODNAME = "deleteWithMeta";
```

Comment:
Update method name

Reply:
good catch thanks! Wish Java had some *simple* and lightweight support for this.